### PR TITLE
Regex test for Youtube URLs

### LINKED ISSUE #10: Write tests to validate Youtube video regex
Write unit tests to verify the correctness of regex used in `Previewer.youtubeLinkToImageUrl` function.

### PR DESCRIPTION
**Pull request by mihir02badkur**
## Description

Added tests to validate Youtube video regex
## Fixes

Fixes #10 by @ArpitShukIa 

## Types of changes

- [X] Bug/Hot fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring Old Code ( non-breaking change replacing already existing code )
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** guidelines in repository.
- [x] My pull request has a descriptive title and is attached to an existing issue.
- [x] I haven't repeated the code and it follows established code style of repository.
- [x] My branch is up-to-date with upstream branch.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -12,6 +12,15 @@
...
         android:theme="@style/Theme.Previewer">
+        <activity
+            android:name=".BitmapVideo"
+            android:exported="true" />
+        <activity
+            android:name=".BitmapYoutube"
+            android:exported="true" />
+        <activity
+            android:name=".BitmapPDF"
+            android:exported="true" />
```
Comment (ArpitShukIa) [inline app/src/main/AndroidManifest.xml:23]: Set `android:exported="false"` for all these activities. We don't want anyone outside this app to be able of launching these activities. Also, change the activity names to `VideoToBitmapActivity`, `YoutubeVideoToBitmapActivity` and `PdfToBitmapActivity`.

File: app/src/main/java/com/cops/iitbhu/previewer/BitmapPDF.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            else
+                Toast.makeText(this, "Unable to generate bitmap!", Toast.LENGTH_SHORT).show()
+        }
+    }
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityBitmapPdfBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.selectFile.setOnClickListener(View.OnClickListener {
```
Comment (ArpitShukIa) [inline app/src/main/java/com/cops/iitbhu/previewer/BitmapPDF.kt:28]: This `View.OnClickListener` is redundant here. You can directly use 
```kotlin
binding.selectFile.setOnClickListener {
    getUriOfPdf.launch(arrayOf("application/pdf"))
}
```

File: app/src/main/java/com/cops/iitbhu/previewer/BitmapYoutube.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityBitmapYoutubeBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.generateThumbnailForYoutube.setOnClickListener {
+            val youtubeLink = binding.youtubeLink.text.toString()
+            if (youtubeLink.isNotEmpty()) {
+                setThumbnail(youtubeLink)
+//                Previewer.setThumbnailFromYouTubeVideoUrl(youtubeLink, binding.img)
```
Comment (ArpitShukIa) [inline app/src/main/java/com/cops/iitbhu/previewer/BitmapYoutube.kt:21]: Add a new button in the activity which will use this non-suspend version to directly set image to ImageView.

File: app/src/main/java/com/cops/iitbhu/previewer/MainActivity.kt [hunk 4]
```diff
@@ -1,71 +1,30 @@
...
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        binding.selectFile.setOnClickListener(View.OnClickListener {
-            getUriOfPdf.launch(arrayOf("application/pdf"))
+        binding.pdf.setOnClickListener(View.OnClickListener {
+            startActivity(Intent(this@MainActivity, BitmapPDF::class.java))
```
Comment (ArpitShukIa) [inline app/src/main/java/com/cops/iitbhu/previewer/MainActivity.kt:19]: Replace `this@MainActivity` with just `this` in all three click listeners.

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -1,118 +1,178 @@
...
+        android:textColor="@color/white"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/youtube_link" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+
+
+
+<!--<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"-->
```
Comment (ArpitShukIa) [inline app/src/main/res/layout/activity_main.xml:63]: Remove all this commented out code from the XML.

File: lib/src/test/java/com/cops/iitbhu/previewer/lib/RegexTest.kt [hunk 6]
```diff
@@ -8,9 +8,47 @@ class RegexTest {
 
     @Test
     fun checkRegexIsValid() {
-        val youtubeLink = "https://www.youtube.com/watch?v=Pur_0KroFR8"
-        val expected = "Pur_0KroFR8"
-        val actual = Previewer.youtubeLinkToImageUrl(youtubeLink)
-        assertEquals(expected, actual)
+        val links= listOf(
```
Comment (ArpitShukIa) [inline lib/src/test/java/com/cops/iitbhu/previewer/lib/RegexTest.kt:11]: Reformat this file once and remove the empty line at the end of this list.

**Review by ArpitShukIa — CHANGES_REQUESTED** [review]
Good job 👍 
Left a few comments. Also remove the first two commits from this PR as they don't belong here.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (mihir02badkur) [reply app/src/main/AndroidManifest.xml:23]: done

File: app/src/main/java/com/cops/iitbhu/previewer/BitmapPDF.kt (hunk 2 above)
Comment (mihir02badkur) [reply app/src/main/java/com/cops/iitbhu/previewer/BitmapPDF.kt:28]: done

File: app/src/main/java/com/cops/iitbhu/previewer/MainActivity.kt (hunk 4 above)
Comment (mihir02badkur) [reply app/src/main/java/com/cops/iitbhu/previewer/MainActivity.kt:19]: done

File: app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (mihir02badkur) [reply app/src/main/res/layout/activity_main.xml:63]: done

File: lib/src/test/java/com/cops/iitbhu/previewer/lib/RegexTest.kt (hunk 6 above)
Comment (mihir02badkur) [reply lib/src/test/java/com/cops/iitbhu/previewer/lib/RegexTest.kt]: done

**Review by ArpitShukIa — APPROVED** [review]
LGTM 👍

**Pull request merged**